requirements/compile.py

```diff
@@ -1,23 +1,20 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.8
+import functools
 import os
+import pathlib
+import shlex
 import subprocess
 import sys
-from pathlib import Path
-
 
 if __name__ == "__main__":
-    os.chdir(Path(__file__).parent)
-    os.environ["CUSTOM_COMPILE_COMMAND"] = "requirements/compile.py"
     common_args = ["-m", "piptools", "compile", "--generate-hashes"] + sys.argv[1:]
-    subprocess.run(
-        ["python3.5", *common_args, "-o", "py35.txt"], check=True,
-    )
-    subprocess.run(
-        ["python3.6", *common_args, "-o", "py36.txt"], check=True,
-    )
-    subprocess.run(
-        ["python3.7", *common_args, "-o", "py37.txt"], check=True,
-    )
-    subprocess.run(
-        ["python3.8", *common_args, "-o", "py38.txt"], check=True,
+    run = functools.partial(
+        subprocess.run,
+        check=True,
+        cwd=pathlib.Path(__file__).parent,
+        env={**os.environ, "CUSTOM_COMPILE_COMMAND": shlex.join(sys.argv)},
```

Comment:
I don't want to pass in `sys.argv` since sometimes I use arguments e.g. `-U` when upgrading everything, but they shouldn't be rendered in the command.